doc/examples/color_exposure/plot_adapt_hist_eq_3d.py

```diff
@@ -0,0 +1,250 @@
+                       ys.flatten()[sel],
+                       zs.flatten()[sel],
+                       c=c, s=0.5 * np.mean(bin_widths),
+                       marker='o', linewidth=0)
+
+
+#############
+# create figure with subplots
+#############
+
+cmap = 'Blues'
+
+fig = plt.figure(figsize=(10, 6))
+axs = [fig.add_subplot(2, 3, i + 1,
+                       projection=Axes3D.name, facecolor="none")
+       for i in range(6)]
+ims = [im_orig, im_orig_he, im_orig_ahe,
+       im_degraded, im_degraded_he, im_degraded_ahe]
+
+# prepare lines for the various boxes to be plotted
+verts = np.array([[i, j, k] for i in [0, 1]
+                  for j in [0, 1] for k in [0, 1]]).astype(np.float32)
+lines = [np.array([i, j]) for i in verts
+         for j in verts if np.allclose(np.linalg.norm(i - j), 1)]
+
+# "render" volumetric data
+for iax, ax in enumerate(axs[:]):
+    plt_render_volume(ims[iax], ax, cmap, 0, 1, [2, 2, 2], 20)
+
+    # plot 3D box
+    rect_shape = np.array(im_orig.shape) + 2
+    for line in lines:
+        ax.plot((line * rect_shape)[:, 0] - 1,
+                (line * rect_shape)[:, 1] - 1,
+                (line * rect_shape)[:, 2] - 1,
+                linewidth=1, color='grey')
+
+# add boxes illustrating the kernels
+ns = np.array(im_orig.shape) // kernel_size - 1
+for iax, ind, rect_shape in zip([1] + [2] * 4,
+                                [[0, 0, 0],
+                                 [ns[0] - 1, ns[1], ns[2] - 1],
+                                 [ns[0], ns[1] - 1, ns[2] - 1],
+                                 [ns[0], ns[1], ns[2] - 1],
+                                 [ns[0], ns[1], ns[2]]],
+                                [np.array(im_orig.shape)]
+                                + [kernel_size] * 4):
+
+    for line in lines:
+        axs[iax].plot(((line + ind) * rect_shape)[:, 0],
+                      ((line + ind) * rect_shape)[:, 1],
+                      ((line + ind) * rect_shape)[:, 2],
+                      linewidth=1.2, color='crimson')
+
+# plot degradation function
+axs[3].scatter(xs=np.arange(len(sigmoid)),
+               ys=np.zeros(len(sigmoid)) + im_orig.shape[1],
+               zs=sigmoid * im_orig.shape[2],
+               s=5,
+               c=get_rgba(sigmoid,
+                          cmap=cmap, vmin=0, vmax=1, alpha=1.)[:, :3])
+
+# subplot aesthetics
+for iax, ax in enumerate(axs[:]):
+
+    # Get rid of panes and axis lines
+    for dim_ax in [ax.w_xaxis, ax.w_yaxis, ax.w_zaxis]:
+        dim_ax.set_pane_color((1., 1., 1., 0.))
+        dim_ax.line.set_color((1., 1., 1., 0.))
+
+    # Make axes limits, see https://github.com/
+    # matplotlib/matplotlib/issues/17172#issuecomment-617546105
+    xyzlim = np.array([ax.get_xlim3d(),
+                       ax.get_ylim3d(),
+                       ax.get_zlim3d()]).T
+    XYZlim = np.asarray([min(xyzlim[0]), max(xyzlim[1])])
+    ax.set_xlim3d(XYZlim)
+    ax.set_ylim3d(XYZlim)
+    ax.set_zlim3d(XYZlim * 0.55)
+
+    try:
```

Comment:
Is this neccessary?

Reply:
I think so, see this thread (commented in the code) https://github.com/matplotlib/matplotlib/issues/17172#issuecomment-617546105. Definitely `napari` will be better for 3D plotting!